src/Tasks/Microsoft.NET.Build.Tasks/GenerateDepsFile.cs

```diff
@@ -103,6 +113,11 @@ protected override void ExecuteCore()
                 .WithReferenceAssembliesPath(FrameworkReferenceResolver.GetDefaultReferenceAssembliesPath())
                 .Build();
 
+            if (compileFilesToSkip.Any() || runtimeFilesToSkip.Any())
+            {
+                dependencyContext = TrimFilesToSkip(dependencyContext);
```

Comment:
Yes, in general the `GenerateDepsFile` task should be pretty implementation free.  It just turns all the inputs into whatever `DependencyContextBuilder` needs.  And then it writes the file.

Please put this logic into `DependencyContextBuilder`.

---
In reply to: [110251089](https://github.com/dotnet/sdk/pull/1052#discussion_r110251089) [](ancestors = 110251089)